src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -914,16 +919,28 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
             status = if (submittedWorkflows.isEmpty) SubmissionStatuses.Done else SubmissionStatuses.Submitted
           )
 
-          if (submission.status == SubmissionStatuses.Submitted) {
-            submissionSupervisor ! SubmissionStarted(workspaceName, submission, userInfo)
+          containerDAO.submissionDAO.save(workspaceContext, submission, txn)
+          val workflowReports = submittedWorkflows.map { workflow =>
+            WorkflowReport(workflow.workflowId, workflow.status, workflow.statusLastChangedDate, workflow.workflowEntity.entityName, workflow.inputResolutions)
           }
 
-          containerDAO.submissionDAO.save(workspaceContext, submission, txn)
-          val workflowReports = for ( (workflow,entityInputs) <- submittedWorkflows.zip(successes) )
-            yield( WorkflowReport(workflow.workflowId,workflow.status,workflow.statusLastChangedDate,entityInputs.entityName,entityInputs.inputResolutions) )
-          RequestComplete(StatusCodes.Created, SubmissionReport(submissionRequest,submission.submissionId,submission.submissionDate,submission.submitter,submission.status,header,workflowReports,failures))
+          RequestComplete(StatusCodes.Created, SubmissionReport(submissionRequest, submission.submissionId, submission.submissionDate, userInfo.userEmail, submission.status, header, workflowReports, failures))
         }
-      }
+    }
+
+    val credFuture = gcsDAO.getUserCredentials(RawlsUser(userInfo))
+
+    submissionFuture.zip(credFuture) map {
+      case (RequestComplete((StatusCodes.Created, submissionReport: SubmissionReport)), Some(credential)) =>
+        if (submissionReport.status == SubmissionStatuses.Submitted) {
+          submissionSupervisor ! SubmissionStarted(workspaceName, submissionReport.submissionId, credential)
+        }
+        RequestComplete(StatusCodes.Created, submissionReport)
+
+      case (somethingWrong, _) => somethingWrong // this is the case where something was not found in withSubmissionParameters
```

Comment:
This is probably good as it is, but just to confirm this is the desired response: when there's something wrong and also no refresh token, it reports the somethingWrong only.